lib/reline/line_editor.rb

```diff
@@ -866,7 +868,7 @@ def editing_mode
       @completion_state = CompletionState::PERFECT_MATCH
     end
     return if result.nil?
-    target, preposing, completed, postposing = result
+    target, preposing, completed, postposing, continuable = result
```

Comment:
`continuable` in calculated twice.
One in complete_internal_proc(`list.count > 1`) and other in this method
```ruby
if list.one?
  @completion_state = CompletionState::PERFECT_MATCH
```

How about using the condition `@completion_state != CompletionState::PERFECT_MATCH` instead of continuable, or set a local variable inside `if list.one?`
```ruby
if list.one?
  @completion_state = CompletionState::PERFECT_MATCH
  perfect_matched_and_need_to_perform_append_char = true # FIXME: shorter name
end
```

Reply:
Thanks for your suggestion! But consider this case when `list` is something like:
```
['foo', 'foo_bar']
```
When we input `foo_[tab]`, it will be completed to `foo_bar`. A completion char should be appended but currently this case will be marked as `MENU_WITH_PERFECT_MATCH`. That's why I separately check `list.count` instead of inside `if list.one? ` case.

After reconsideration, I think maybe it's OK to treat the above case as a `PERFECT_MATCH`?

Reply:
Treating it as PERFECT_MATCH makes sense to me :+1: